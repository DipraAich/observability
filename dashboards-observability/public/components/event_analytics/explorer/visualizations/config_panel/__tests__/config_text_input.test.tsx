/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { TextInputFieldItem } from '../config_panes/config_controls/config_text_input';

describe('Config text input component', () => {
  configure({ adapter: new Adapter() });
  it('Renders TextInputFieldItem with empty data', async () => {
    const wrapper = mount(
      <TextInputFieldItem name={''} title={''} currentValue={''} handleInputChange={jest.fn()} />
    );

    wrapper.update();
    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('Renders TextInputFieldItem with data', async () => {
    const wrapper = mount(
      <TextInputFieldItem
        name={'seriesUnits'}
        title={'Series'}
        currentValue={'units'}
        handleInputChange={jest.fn()}
      />
    );

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
