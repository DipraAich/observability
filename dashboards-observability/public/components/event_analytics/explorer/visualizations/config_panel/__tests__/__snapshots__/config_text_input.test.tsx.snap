// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config text input component Renders TextInputFieldItem with data 1`] = `
<TextInputFieldItem
  currentValue="units"
  handleInputChange={[MockFunction]}
  name="seriesUnits"
  title="Series"
>
  <EuiTitle
    size="xxs"
  >
    <h3
      className="euiTitle euiTitle--xxsmall"
    >
      Series
    </h3>
  </EuiTitle>
  <EuiSpacer
    size="s"
  >
    <div
      className="euiSpacer euiSpacer--s"
    />
  </EuiSpacer>
  <EuiFieldText
    data-test-subj="valueFieldText"
    fullWidth={true}
    id="random_html_id"
    name="seriesUnits"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Series"
    value="units"
  >
    <EuiFormControlLayout
      fullWidth={true}
      inputId="random_html_id"
    >
      <div
        className="euiFormControlLayout euiFormControlLayout--fullWidth"
      >
        <div
          className="euiFormControlLayout__childrenWrapper"
        >
          <EuiValidatableControl>
            <input
              className="euiFieldText euiFieldText--fullWidth"
              data-test-subj="valueFieldText"
              id="random_html_id"
              name="seriesUnits"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Series"
              type="text"
              value="units"
            />
          </EuiValidatableControl>
          <EuiFormControlLayoutIcons />
        </div>
      </div>
    </EuiFormControlLayout>
  </EuiFieldText>
</TextInputFieldItem>
`;

exports[`Config text input component Renders TextInputFieldItem with empty data 1`] = `
<TextInputFieldItem
  currentValue=""
  handleInputChange={[MockFunction]}
  name=""
  title=""
>
  <EuiTitle
    size="xxs"
  >
    <h3
      className="euiTitle euiTitle--xxsmall"
    />
  </EuiTitle>
  <EuiSpacer
    size="s"
  >
    <div
      className="euiSpacer euiSpacer--s"
    />
  </EuiSpacer>
  <EuiFieldText
    data-test-subj="valueFieldText"
    fullWidth={true}
    id="random_html_id"
    name=""
    onBlur={[Function]}
    onChange={[Function]}
    placeholder=""
    value=""
  >
    <EuiFormControlLayout
      fullWidth={true}
      inputId="random_html_id"
    >
      <div
        className="euiFormControlLayout euiFormControlLayout--fullWidth"
      >
        <div
          className="euiFormControlLayout__childrenWrapper"
        >
          <EuiValidatableControl>
            <input
              className="euiFieldText euiFieldText--fullWidth"
              data-test-subj="valueFieldText"
              id="random_html_id"
              name=""
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=""
              type="text"
              value=""
            />
          </EuiValidatableControl>
          <EuiFormControlLayoutIcons />
        </div>
      </div>
    </EuiFormControlLayout>
  </EuiFieldText>
</TextInputFieldItem>
`;
